microsoft-azure-storage/src/com/microsoft/azure/storage/core/Canonicalizer.java

```diff
@@ -251,6 +256,11 @@ protected static String getCanonicalizedResource(final java.net.URL address, fin
         final StringBuilder canonicalizedResource = new StringBuilder(resourcepath.toString());
 
         // query parameters
+        if (address.getQuery()!= null  && !address.getQuery().contains("=")) {
+            //no query params. Safe to return empty
+            return canonicalizedResource.toString();
+        }
```

Comment:
I did a full test pass and everything passed. Running it now in debug mode because I want to know when this is happening. Do you have an example API that you know this saves time for?